### PR TITLE
Add ability to overlay project.ptx from the commandline

### PR DESCRIPTION
**Pull request by siefkenj**
As discussed on the mailing list, this PR adds the ability to overlay settings in `project.ptx` from the command line.

For example, if your `project.ptx` file didn't have custom xsl or you wanted to override the custom xsl file, you could run

```
pretext build -P targets.target.xsl "my/custom/path.xsl"
```

If you wanted to change an attribute, you can use the `@` syntax.
```
pretext build -P targets.target@pdf-method pdflatex
```

Multiple instances of `-P`/`--project-ptx-override` can be applied. The largest limitation is that it is based on a simple selector language, using `.` to indicate node hierarchy. There is no way to distinguish between siblings with the same name, so if your tree is `<a><b>hi</b><b>there</b></a>`, using `-P a.b foo` will result in `<a><b>foo</b><b>foo</b></a>`. I think this is a reasonable limitation. If you want full control over the `project.ptx` file, you should create one matching your specifications, rather than using a complex query language on the command line.

The benefit of this feature is that it lets you test out small modifications to `project.ptx` without any source modifications. E.g., if you are working in a git-tracked repo and you wanted to test a different output format but didn't want to worry about accidentally committing a changed `project.ptx` file, this would let you specify `-P targets.target.format ...` on the command line :-).

This may be a partial solution for #327 
This PR depends on #366

### CODE REVIEW

**Comment by oscarlevin** [comment]
This will be a nice addition, and help quite a bit with getting the CodeChat working (it can call pretext build web with a custom `html-with-mapping` format).

Should we also make this option available for `pretext generate`?

**Comment by siefkenj** [comment]
@oscarlevin Good point! I have added the functionality to `pretext generate`.

**Comment by StevenClontz** [comment]
Is there a non-manual way to deal with the merge conflicts?

**Comment by siefkenj** [comment]
@StevenClontz Not that I know of...but I took care of all the conflicts. Should be reviewable now!

File: pretext/cli.py
Comment (StevenClontz) [inline pretext/cli.py:346]: CLI convention is to use lowercase flags

File: pretext/cli.py
Comment (StevenClontz) [inline pretext/cli.py:428]: `-p`

**Review by StevenClontz — CHANGES_REQUESTED** [review]

File: pretext/cli.py
Comment (siefkenj) [reply pretext/cli.py:346]: Sure, I'll switch it. I put it as a capital because I thought it was an "advanced" option, that most users wouldn't use, so we could save lower-case letters for common options.

File: pretext/config/xml_overlay.py
Comment (siefkenj) [reply pretext/config/xml_overlay.py:6]: The trouble is that new lines get carried over when `--help` is used. I didn't want them to mess up the docs, but I also wanted the doc string to be easily editable. I'll move to utils. Maybe I should change the name to `format_for_click_help`?

**Comment by siefkenj** [comment]
Very weird...I don't know why the CI isn't running on my latest commit. Does that make sense to you @StevenClontz ?

**Comment by rbeezer** [comment]
I can't fetch a PR right now.  Maybe GH is borked...

**Comment by StevenClontz** [comment]
Too lazy to do UTC conversion but it seems likely there was an issue now
resolved: https://www.githubstatus.com/history


Steven Clontz
https://clontz.org - ***@***.***

**Comment by StevenClontz** [comment]
Going ahead and approving assuming tests pass. Not sure the reasoning for a `config` module for this to live in but don't want to hold this up further.

**Review by StevenClontz — APPROVED** [review]

**Pull request merged**
